planner/core/task.go

```diff
@@ -716,12 +714,8 @@ func (p *PhysicalHashJoin) convertJoinKeyForTiFlashIfNeed(lTask, rTask *mppTask)
 			rCast := expression.BuildCastFunction(p.ctx, rKey, cType)
 			rKey = appendExpr(rProj, rCast)
 		}
-		if lMask[i] || rMask[i] {
```

Comment:
Why removes this check?

Reply:
as HashJoin [ToPB](https://github.com/pingcap/tidb/blob/3f6dfe91daab7acb178ad24941dfca1745993afd/planner/core/plan_to_pb.go#L335) does not serialize the `p.EqualConditions`, it is enough that we only convert the data type of partition keys.